src/lib.rs

```diff
@@ -251,40 +258,76 @@ impl PerformanceData {
             },
             SetTimeFlags::Default,
         );
+        let timer_pollfd = PollFd::new(tfd.as_fd(), PollFlags::POLLIN);
+
+        // SignalFd
+        let mut mask = SigSet::empty();
+        mask.add(signal::SIGINT);
+        mask.add(signal::SIGTERM);
+        mask.thread_block()?;
+        let sfd = SignalFd::with_flags(&mask, SfdFlags::SFD_NONBLOCK)?;
+        let signal_pollfd = PollFd::new(sfd.as_fd(), PollFlags::POLLIN);
+
+        let mut poll_fds = [timer_pollfd, signal_pollfd];
+
         while current <= end {
             aperf_collect_data.time = TimeEnum::DateTime(Utc::now());
             aperf_collect_data.data = HashMap::new();
-            let ret = tfd.read();
-            if ret > 1 {
-                error!("Missed {} interval(s)", ret - 1);
+            if poll(&mut poll_fds, PollTimeout::NONE)? <= 0 {
+                error!("Poll error.");
+            }
+            if let Some(ev) = poll_fds[0].revents() {
```

Comment:
I think it's a bit nicer here to use timer_pollfd instea dof poll_fds[0]

And if possible, I'd change the argument to poll() to be the literal [timer_pollfd, signal_pollfd] (if it will let you, I'm not sure it will), to emphasize the unimportance of that array

Reply:
Yeah. It won't let me do that.

Reply:
I'm sure there's a way to make this less fragile